### PR TITLE
[SPARK-40160][PYTHON][DOCS] Make pyspark.broadcast examples self-contained

### PR DESCRIPTION
**Pull request by dcoliversun**
### What changes were proposed in this pull request?

This PR proposes to improve the examples in `pyspark.broadcast` by making each example self-contained with a brief explanation and a bit more realistic example.

### Why are the changes needed?

To make the documentation more readable and able to copy and paste directly in PySpark shell.

### Does this PR introduce _any_ user-facing change?

Yes, it changes the documentation.

### How was this patch tested?

Manually ran each doctest.

### CODE REVIEW

File: python/pyspark/broadcast.py [hunk 1]
```diff
@@ -149,6 +147,18 @@ def __init__(
...
+        The protocol pickle is HIGHEST_PROTOCOL.
+
+        Parameters
+        ----------
+        value : T
+            Value to write.
+
+        f : :class:`BinaryIO`
+            File or socket where the pickled value will be stored.
+        """
```
Comment (HyukjinKwon) [inline python/pyspark/broadcast.py:161]: Could we maybe add an example with dump and load? E.g. with tempfile.TemporaryDirectory

Comment (dcoliversun) [reply python/pyspark/broadcast.py:175]: Updated

File: python/pyspark/broadcast.py [hunk 2]
```diff
@@ -246,9 +349,20 @@ def clear(self) -> None:
...
 
 
-if __name__ == "__main__":
+def _test() -> None:
     import doctest
+    from pyspark.sql import SparkSession
+    import pyspark.broadcast
+
+    globs = pyspark.broadcast.__dict__.copy()
+    spark = SparkSession.builder.master("local[4]").appName("pyspark.broadcast tests").getOrCreate()
```
Comment (HyukjinKwon) [inline python/pyspark/broadcast.py:358]:
```suggestion
    spark = SparkSession.builder.master("local[4]").appName("broadcast tests").getOrCreate()
```
to be consistent

File: python/pyspark/broadcast.py [hunk 3]
```diff
@@ -195,7 +282,15 @@ def unpersist(self, blocking: bool = False) -> None:
         Parameters
         ----------
         blocking : bool, optional
```
Comment (HyukjinKwon) [inline python/pyspark/broadcast.py:284]:
```suggestion
        blocking : bool, optional, default False
```

File: python/pyspark/broadcast.py [hunk 4]
```diff
@@ -195,7 +282,15 @@ def unpersist(self, blocking: bool = False) -> None:
...
-            Whether to block until unpersisting has completed
+            Whether to block until unpersisting has completed, default `False`.
```
Comment (HyukjinKwon) [inline python/pyspark/broadcast.py:285]:
```suggestion
            Whether to block until unpersisting has completed.
```

File: python/pyspark/broadcast.py [hunk 5]
```diff
@@ -214,7 +309,15 @@ def destroy(self, blocking: bool = False) -> None:
         Parameters
         ----------
         blocking : bool, optional
```
Comment (HyukjinKwon) [inline python/pyspark/broadcast.py:311]:
```suggestion
        blocking : bool, optional, default False
```

File: python/pyspark/broadcast.py [hunk 6]
```diff
@@ -214,7 +309,15 @@ def destroy(self, blocking: bool = False) -> None:
...
-            Whether to block until unpersisting has completed
+            Whether to block until unpersisting has completed, default `False`.
```
Comment (HyukjinKwon) [inline python/pyspark/broadcast.py:312]:
```suggestion
            Whether to block until unpersisting has completed.
```

File: python/pyspark/broadcast.py [hunk 7]
```diff
@@ -246,9 +349,20 @@ def clear(self) -> None:
         self._registry.clear()
 
 
-if __name__ == "__main__":
+def _test() -> None:
```
Comment (HyukjinKwon) [inline python/pyspark/broadcast.py:352]:
```suggestion
def _test():
```

This is fine. PySpark only types the public API only :-).

**Review by HyukjinKwon — APPROVED** [review]
LGTM, thanks for working on this.

File: python/pyspark/broadcast.py (hunk 7 above)
Comment (dcoliversun) [reply python/pyspark/broadcast.py:352]: without ` -> None`, python lint test failed

Comment (HyukjinKwon) [reply python/pyspark/broadcast.py:352]: interesting ..

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**